File: app/src/main/java/com/pirateman/exercisemanager/exercise/ExerciseEditActivity.java
package com.pirateman.exercisemanager.exercise;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.util.Log;
import android.view.KeyEvent;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.pirateman.exercisemanager.R;

import java.util.concurrent.Executor;
import java.util.concurrent.Executors;

public class ExerciseEditActivity extends AppCompatActivity implements TextWatcher {

    private Exercise exercise;
    OnUpdateExerciseListener listener;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_exercise_edit);
        listener = (OnUpdateExerciseListener) this;

        Bundle data = getIntent().getExtras();

        String tvName = data.getString("tvName", "Name");
        String tvMethod = data.getString("tvMethod", "Method");
        String tvMuscleGroup = data.getString("tvMuscleGroup", "Muscle Group");

        final EditText edtName = findViewById(R.id.edtName);
        edtName.setText(tvName);
        final EditText edtMethod = findViewById(R.id.edtMethod);
        edtMethod.setText(tvMethod);
        final EditText edtMuscleGroup = findViewById(R.id.edtMuscleGroup);
        edtMuscleGroup.setText(tvMuscleGroup);

        edtName.addTextChangedListener(this);


        Button btnEditSubmit = findViewById(R.id.btnEditSubmit);

        btnEditSubmit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                listener.updateExercise(new Exercise(edtName.getText().toString(), edtMuscleGroup.getText().toString(), edtMethod.getText().toString()));
            }
        });

    }

    @Override
    public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {

    }

    @Override
    public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {

    }

    @Override
    public void afterTextChanged(Editable editable) {

    }
}

File: app/src/main/java/com/pirateman/exercisemanager/exercise/ExerciseEditActivity.kt
package com.pirateman.exercisemanager.exercise

import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import com.pirateman.exercisemanager.R

class ExerciseEditActivity : AppCompatActivity(), TextWatcher {
    private val exercise: Exercise? = null
    var listener: OnUpdateExerciseListener? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_exercise_edit)
        listener = this as OnUpdateExerciseListener
        val data = intent.extras
        val tvName = data!!.getString("tvName", "Name")
        val tvMethod = data.getString("tvMethod", "Method")
        val tvMuscleGroup = data.getString("tvMuscleGroup", "Muscle Group")
        val edtName = findViewById<EditText>(R.id.edtAddName)
        edtName.setText(tvName)
        val edtMethod = findViewById<EditText>(R.id.edtAddMethod)
        edtMethod.setText(tvMethod)
        val edtMuscleGroup = findViewById<EditText>(R.id.edtAddMuscleGroup)
        edtMuscleGroup.setText(tvMuscleGroup)
        edtName.addTextChangedListener(this)
        val btnEditSubmit = findViewById<Button>(R.id.btnSubmitNewExercise)
        btnEditSubmit.setOnClickListener { listener!!.updateExercise(Exercise(edtName.text.toString(), edtMuscleGroup.text.toString(), edtMethod.text.toString())) }
    }

    override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
    override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
    override fun afterTextChanged(editable: Editable) {}
}